### PR TITLE
Remove non-ascii alphanumericals when generating heading id

### LINKED ISSUE #224: Remove any characters that aren't valid when generating a heading id.
Remove any characters that aren't [a valid css identifier](https://www.w3.org/TR/CSS21/syndata.html#value-def-identifier) when generating the heading id.

It appears only whitespace is removed currently. 
https://github.com/Financial-Times/o-overlay/blob/81e779ae0af90616aeedb796ef42f3d9beb8855e/src/js/overlay.js#L246

Relates to https://github.com/Financial-Times/o-overlay/pull/223

### PR DESCRIPTION
**Pull request by chee**
closes #224 imho

### CODE REVIEW

File: src/js/overlay.js [hunk 1]
```diff
@@ -243,7 +276,12 @@ class Overlay {
 
 		if (this.opts.heading) {
 			const heading = document.createElement('header');
-			const headingId = this.opts.heading.title.replace(/\s+/g, '-').toLowerCase();
+			const headingId = this.opts.heading.title
+				.toLowerCase()
+				// replace non-ascii alphanums with hyphens
+				.replace(/[^a-z0-9-]/g, '-')
+				// replace repeated hyphens with a single hyphen
+				.replace(/[\s-]+/g, '-');
```
Comment (JakeChampion) [inline src/js/overlay.js:284]: 👍

**Review by JakeChampion — APPROVED** [review]

**Comment by chee** [comment]
the bower build/test passes.

the npm build is failing in CI, so i'm running it locally to see if i can understand why.

**Comment by notlee** [comment]
This looks good to me but I wonder if it should be treated as a breaking change? There's a small chance users have relied on the generated id.

**Comment by chee** [comment]
there seems to be a bug in one of the many complex parts of our build toolchain, for some reason it is trying to run `main.js` instead of `browser.js`

**Comment by chee** [comment]
i think it's this? https://github.com/Financial-Times/origami-build-tools/blob/fd3573f9689b1672b5cf1020375f4fa90120a595/lib/tasks/build-js.js#L34

**Comment by chee** [comment]
> there seems to be a bug in one of the many complex parts of our build toolchain, for some reason it is trying to run `main.js` instead of `browser.js`

oh, it might be just that the demos are literally asking for it

File: demos/src/sliding-notification.js [hunk 2]
```diff
@@ -1,6 +1,6 @@
-import Overlay from '../../main.js';
+import Overlay from '../..';
```
Comment (JakeChampion) [inline demos/src/sliding-notification.js:1]: Nice work finding this!

**Comment by origamiserviceuser** [comment]
o-overlay bundle size difference from 3.0.9 to a6a692b7242a00b8a113543266a162dff012b11f
No significant bundle size differences found.

**Comment by chee** [comment]
yay!

**Comment by chee** [comment]
> This looks good to me but I wonder if it should be treated as a breaking change? There's a small chance users have relied on the generated id.
> 
> Edit: That feels very unlikely and a poor choice since its based on text content. Let's release as a patch +1

i think you're right :)

**Comment by origamiserviceuser** [comment]
:tada: This PR is included in version v3.0.10 :tada:

**Pull request merged**